src/dune_rules/workspace.ml

```diff
@@ -384,7 +406,8 @@ module Context = struct
           "instrument_with"
           (Dune_lang.Syntax.since syntax (2, 7) >>> repeat Lib_name.decode)
       and+ loc = loc
-      and+ merlin = field_b "merlin" in
+      and+ merlin = field_b "merlin"
+      and+ generate_merlin_rules = field_b "generate_merlin_rules" in
```

Comment:
should this have a `since 3.16` check?

Reply:
👍 done in https://github.com/ocaml/dune/pull/10324/commits/d384104f3f2872b1c16c2f5a61c556292675e190.